src/common/utils/deploy.utils.ts

```diff
@@ -99,4 +102,12 @@ export default class DeployUtils {
     }
     return interfaces_map;
   }
+
+  static async hasClusterDeployment(app: AppService, cluster: Cluster): Promise<boolean> {
+    if (cluster && !cluster.properties.is_shared && !cluster.properties.is_managed) {
+      const cluster_deployment: Deployment = (await app.api.get(`clusters/${cluster.id}/apps`)).data;
+      return !(cluster_deployment && cluster_deployment.action === 'delete');
```

Comment:
This check looks like it will catch a case of the cluster apps being installed then uninstalled, but not a case of the cluster apps never being installed in the first place